invenio_stats/tasks.py

```diff
@@ -38,3 +43,240 @@ def process_events(event_types):
     for e in event_types:
         results.append((e, current_stats.events[e].processor.run()))
     return results
+
+
+@shared_task
+def aggregate_event(event):
+    """Calculate file downloads for today."""
+    file_download_aggregator = StatAggregator(
+        client=current_search_client,
+        event='file-download')
+    file_download_aggregator.run()
+
+
+class StatPreprocessor(object):
+    """Preprocessor class.
+
+    Subclass this class in order to provide specific statistics calculations.
+    """
+
+    def run(self):
+        """Run calculations."""
+        pass
+
+
+class StatAggregator(StatPreprocessor):
```

Comment:
Just a tiny comment - I think API classes like this one fits better in their own Python modules (or in an ``api.py``), so that ``tasks.py`` is kept as primarily a controller.